### DISCUSSION TITLE
How would you dynamically update unit of measurement?

### DISCUSSION THREAD

**Discussion by RostumAhady** [angularsen/UnitsNet#952] [Q&A]
I'm working on a project involved with recipes. So there's lots of different units used. One thing I'd like to do is dynamically display a larger unit of measurement when a user is increasing serving size, without hard coding the breakpoints for every specific unit of measurement (example, 1000g to 1kg). Does something like this exist in UnitsNet already?

**Comment by angularsen** [comment]
@RostumAhady Hi, there is no such functionality in the library today. Humanizer is good at this for things like relative time ("5 minutes ago"), so at first glance I would agree it sounds useful to add similar functionality to Units.NET.

However, I'm not sure how to implement this in a way that works for most people:
- We have competing units of similar scale. Humanizer has to pick between seconds, minutes, hours etc, but we have to choose between kilograms, pounds, etc.
- Units are often domain specific. For baking, people often use units like cup and teaspoon that would be awkward in other domains.
- If the value increases from 999 grams to 1001 grams, should it then show as 1001 grams or 1.001 kg?

Do you have an idea for how you would like the breakpoints to work?
